### PR TITLE
Release docs

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update some documentation for the 2.3.1 release and merge in changes to master

### Proposed changes

- doc update
- some cmake formatting
- merge recent master changes

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/HELICS/HELICS/pullRequest?prid=4545591&bid=15294151)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,29 +7,35 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 
-## \[2.3.1\] ~ In Development
+## \[2.3.1\] - 2019-11-22
 Bug Fixes and some code refactoring, pkg-config files have been added to the installs
 ### Changed
 -   Default installation path for MSYS2 is now configured to be part of the system path, typically `/mingw64/`  or `/mingw32/`
 -   `HELICS_ENABLE_SLOW_PACKAGING_TESTS` renamed to `HELICS_ENABLE_SUBPROJECT_TESTS` to better reflect usage
 -   filesystem library updated to clear up some warnings
+-   The CI system now runs Xcode9 as the oldest release
+-   Automatic build system was changed to use scripts
```
Comment (nightlark) [inline CHANGELOG.md]:
```suggestion
-   Automatic releases build system was changed to use scripts 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,29 +7,35 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 -   Default installation path for MSYS2 is now configured to be part of the system path, typically `/mingw64/`  or `/mingw32/`
 -   `HELICS_ENABLE_SLOW_PACKAGING_TESTS` renamed to `HELICS_ENABLE_SUBPROJECT_TESTS` to better reflect usage
 -   filesystem library updated to clear up some warnings
+-   The CI system now runs Xcode9 as the oldest release
+-   Automatic build system was changed to use scripts 
 
 ### Fixed
 -   Some documentation links in the docs
 -   Missing `helics-enums.h` header from the install if `HELICS_BUILD_CXX_SHARED_LIB` was not enabled
+-  ZMQ install locations on Linux and MacOS if ZMQ is used as a subproject without the HELICS_USE_ZMQ_STATIC_LIB
```
Comment (nightlark) [inline CHANGELOG.md]:
```suggestion
-  ZMQ install locations on Linux and macOS if ZMQ is used as a subproject without the HELICS_USE_ZMQ_STATIC_LIB option enabled
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,29 +7,35 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+-   The linux shared library release build so it is compatible with a larger number of systems including older ones back to glibc 2.12. 
+-  Fix some issues with using the STATIC_STANDARD_LIB CMake option
```
Comment (nightlark) [inline CHANGELOG.md]:
```suggestion
-  Fix some documentation and issues with using the STATIC_STANDARD_LIB CMake option
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,29 +7,35 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
   
 ### Added
 -   CMake option for `HELICS_DISABLE_ASIO` to completely remove the use the ASIO library, turns off the UDP, and TCP core types, all real-time capabilities, and timeout and heartbeat detection for cores and brokers.  ASIO doesn't support all version of cygwin.  
 -   pkg-config files for the shared libraries are now installed to `<prefix>/lib/pkg-config` on unix like systems  
 -   Tests and CI builds for installed CMake package files and pkg-config files
 
 ### Deprecated
+-  trying to install on linux/macos systems with cmake older than 3.13 and ZMQ used as a subproject with the shared library is no longer supported.  It is likely this use scenario was broken before, now it produces a warning.
```
Comment (nightlark) [inline CHANGELOG.md]:
```suggestion
-  Trying to install on linux/macos systems with cmake older than 3.13 and ZMQ used as a subproject with the shared library is no longer supported.  It is likely this use scenario was broken before, now it produces a warning.  
```

File: docs/ROADMAP.md [hunk 5]
```diff
@@ -13,18 +13,21 @@ This document contains tentative plans for changes and improvements of note in u
 -   Broker-server continued development and support for tcp/udp cores
 -   Additional package manager integration
 -   Performance improvements and tests
+-   Multi-node benchmark tests
 -   messageObject callbacks into the C library
+-   Improved installers for MacOS
```
Comment (nightlark) [inline docs/ROADMAP.md]:
```suggestion
-   Improved installers for macOS
```

File: docs/ROADMAP.md [hunk 6]
```diff
@@ -13,18 +13,21 @@ This document contains tentative plans for changes and improvements of note in u
...
 -   Automated code formatting in github
 -   Separate out networking library
 -   Debugging tools
--   SSL capable core (unlikely)
 -   Split core library between comms layer components and actual core API
--   Plugin architecture for user defined cores (unlikely)
 -   Drop tested support for Xcode 8
```
Comment (nightlark) [inline docs/ROADMAP.md]:
```suggestion
```

File: docs/developer-guide/continuous-integration.md [hunk 7]
```diff
@@ -17,18 +17,40 @@ Push tests run on all pushes to any branch in the main repo, there are 4 tests t
 ### PR tests and develop branch Tests
 Pull request tests run on every pull request to develop or master, In addition to the previous 4 tests 2 additional tests are run.
  - Clang 3.6: which is the oldest fully supported clang compiler, with boost 1.58 (Build only)
- - XCode 8.3:  which is the oldest fully supported Xcode version (not for PRs to develop)
+ - XCode 10.2:
```
Comment (nightlark) [inline docs/developer-guide/continuous-integration.md]:
```suggestion
 - XCode 10.2
```

File: docs/developer-guide/continuous-integration.md [hunk 8]
```diff
@@ -17,18 +17,40 @@ Push tests run on all pushes to any branch in the main repo, there are 4 tests t
 ### PR tests and develop branch Tests
 Pull request tests run on every pull request to develop or master, In addition to the previous 4 tests 2 additional tests are run.
```
Comment (nightlark) [inline docs/developer-guide/continuous-integration.md]:
```suggestion
Pull request tests run on every pull request to develop or master. In addition to the previous 4 tests 2 additional tests are run.
```

File: docs/developer-guide/continuous-integration.md [hunk 9]
```diff
@@ -17,18 +17,40 @@ Push tests run on all pushes to any branch in the main repo, there are 4 tests t
...
 
 ## Appveyor tests
-  - MSVC 2015 cmake 3.13,  python and java builds and tests, packaging tests for windows
+  - MSVC 2015 cmake 3.13,  python and java builds 
 
 ## Azure tests
 PRs and commits to the master and develop branches that pass the tests on Travis will trigger builds on Azure for several other HELICS related repositories (such as HELICS-Examples). The result of the builds for those repositories will be reported as a comment on the PR (if any) that triggered the build.
+
+On the Primary HELICS repository there are 4 Azure builds:
+-    MSVC2015 64bit Build and test, choclatey swig/boost
```
Comment (nightlark) [inline docs/developer-guide/continuous-integration.md]:
```suggestion
-    MSVC2015 64bit Build and test, chocolatey swig/boost
```

File: docs/developer-guide/continuous-integration.md [hunk 10]
```diff
@@ -17,18 +17,40 @@ Push tests run on all pushes to any branch in the main repo, there are 4 tests t
...
+-    MSVC2017 32bit Build and test
+-    MSVC2017 64bit Build and test with Java
+-    MSVC2019 64bit Build and test with Java 
+
+## Circle CI
+All PR's and branches trigger a set of Builds using Docker images on Circle-CI.
```
Comment (nightlark) [inline docs/developer-guide/continuous-integration.md]:
```suggestion
All PR's and branches trigger a set of builds using Docker images on Circle-CI.  
```

File: docs/developer-guide/continuous-integration.md [hunk 11]
```diff
@@ -17,18 +17,40 @@ Push tests run on all pushes to any branch in the main repo, there are 4 tests t
...
+All PR's and branches trigger a set of Builds using Docker images on Circle-CI.  
+-   Octave tests - tests the Octave interface and runs some tests
+-   Clang-MSAN - runs the clang memory sanitizer
+-   Cland-ASAN - runs the clang address sanitizer and undefined behavior sanitizer
+-   Clang-TSAN - runs the clang thread santizer
+-   install1  - build and install and link with the C shared library, C\++ shared library, C\++98 library and C\++ apps library, and run some tests linking to the installed libraries
+-   install2  - build and install and link with the C shared library, and C\++98 library only and run some tests linking with the installed library
+
+## Drone
+-  64 bit build and 32 bit builds
```
Comment (nightlark) [inline docs/developer-guide/continuous-integration.md]:
```suggestion
-  64 bit and 32 bit builds on ARM processors
```

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/HELICS/HELICS/pullRequest?prid=4545591&bid=15294151)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/HELICS/HELICS/pullRequest?prid=4545591&bid=15294151)

**Pull request merged**